arbitrator/include/capabilities_interface.tpp

```diff
@@ -23,38 +23,96 @@
 #include <functional>
 #include <carma_planning_msgs/srv/plan_maneuvers.hpp>
 #include <rclcpp/exceptions/exceptions.hpp>
-
+#include <chrono>
+#include <thread>
 namespace arbitrator
 {
+    constexpr auto MAX_RETRY_ATTEMPTS {10};
+
     template<typename MSrvReq, typename MSrvRes>
     std::map<std::string, std::shared_ptr<MSrvRes>>
         CapabilitiesInterface::multiplex_service_call_for_capability(const std::string& query_string, std::shared_ptr<MSrvReq> msg)
     {
         std::map<std::string, std::shared_ptr<MSrvRes>> responses;
-
-        for (const auto & topic : get_topics_for_capability(query_string))
+        bool topics_call_successful = false;
+        int retry_attempt = 0;
+        std::vector<std::string> detected_topics;
+        while (!topics_call_successful && retry_attempt < MAX_RETRY_ATTEMPTS)
         {
             try {
-                using std::literals::chrono_literals::operator""ms;
+            detected_topics = get_topics_for_capability(query_string);
+            topics_call_successful = true;
+            if (retry_attempt > 0)
+            {
+                RCLCPP_INFO_STREAM(rclcpp::get_logger("arbitrator")," Service call to get available strategic plugins successfully finished after retrying: " << retry_attempt);
+            }
+            } catch (const rclcpp::exceptions::RCLError& error) {
+                RCLCPP_WARN_STREAM(rclcpp::get_logger("arbitrator"),
+                            "Could not make a service call to get available strategic service topic names, with detected error: " << error.what() <<". So retrying, attempt no: " << retry_attempt);
+                retry_attempt ++;
+            }
+
+        }
+
+        if (!topics_call_successful)
+        {
+            RCLCPP_ERROR_STREAM(rclcpp::get_logger("arbitrator"),
+                            "Failed to retrieve available strategic plugin service topics! Returning empty maneuver responses");
+            return responses;
+        }
+
+        std::vector<std::string> topics_to_retry = detected_topics;
+        std::vector<std::string> current_topics_to_check;
+        retry_attempt = 0;
+        while (!topics_to_retry.empty() && retry_attempt < MAX_RETRY_ATTEMPTS)
+        {
+            current_topics_to_check = topics_to_retry;
+            topics_to_retry.clear();
+            for (const auto & topic : current_topics_to_check)
+            {
+                try {
+                    using std::literals::chrono_literals::operator""ms;
+
+                    if (registered_strategic_plugins_.find(topic) == registered_strategic_plugins_.end())
```

Comment:
I think using `count()` would be better here since we don't need an iterator in this context.
```cpp
if (registered_strategic_plugins_.count(topic) == 0)
```

Reply:
changed